lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -19,120 +20,186 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+
+Function CheckModified-File($url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy) {
```

Comment:
Just pass in `$proxy` and set it to null when you create if instead of putting the if statement twice in each function

Reply:
Finally got round to testing this and you are correct with your assumptions. If no proxy is set under the `Proxy` property it will use the system default. If `$null` is set it will ignore the system default and finally you can set one to override the system default with `System.Net.WebProxy`.

I still think this can be cleared up a tiny bit but won't stop that from merging.